pennylane/pauli/pauli_arithmetic.py

```diff
@@ -235,6 +224,21 @@ def __matmul__(self, other):
 
         return PauliWord(result), coeff
 
+    def __matmul__(self, other):
+        """Multiply two Pauli words together using the matrix product if wires overlap
+        and the tensor product otherwise.
+
+        Empty Pauli words are treated as the Identity operator on all wires.
+
+        Args:
+            other (PauliWord): The Pauli word to multiply with
+
+        Returns:
+            PauliSentence: coeff * new_word
```

Comment:
I think the way it is written now is better because coeff * PauliWord does return a Paulisentence, the dict structure looks odd here. Alternatively we could do
```suggestion
            PauliSentence: PauliSentence({new_word: coeff})
```

Reply:
I think either of your suggestions make sense, on second thought the `PauliSentence` also prints in the format `coeff * pauli_word`, so maybe just leave as is. `PauliSentence({new_word: coeff})` also seems good 🤷‍♀️

Reply:
Gonna leave as is then 👍